redfish-core/lib/assembly.hpp

```diff
@@ -209,6 +209,28 @@ void getAssemblyLocationCode(
     });
 }
 
+void getAssemblyPresence(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                         const auto& serviceName, const auto& assembly,
+                         const auto& assemblyIndex)
+{
+    sdbusplus::asio::getProperty<bool>(
+        *crow::connections::systemBus, serviceName, assembly,
+        "xyz.openbmc_project.Inventory.Item", "Present",
+        [asyncResp, assemblyIndex](const boost::system::error_code ec,
+                                   const bool value) {
+        if (ec)
+        {
+            BMCWEB_LOG_DEBUG("DBUS response error: {}", ec.value());
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        nlohmann::json& assemblyArray = asyncResp->res.jsonValue["Assemblies"];
+        nlohmann::json& assemblyData = assemblyArray.at(assemblyIndex);
+
+        assemblyData["Status"]["State"] = value ? "Enabled" : "Absent";
```

Comment:
Is this having a default value or empty - e.g. when nothing in dbus?
In other places, it looks like - it is set to "Enabled" first, and then set it as "Absent" if value == false.

For example,
- https://github.com/ibm-openbmc/bmcweb/blob/44547a092b185ee76d6a6e3f5ab2a5814c3c8a47/redfish-core/lib/fabric_adapters.hpp#L302
- https://github.com/ibm-openbmc/bmcweb/blob/44547a092b185ee76d6a6e3f5ab2a5814c3c8a47/redfish-core/lib/fabric_adapters.hpp#L147

Should this be doing the same like?

L265:
```
        item["MemberId"] = std::to_string(assemblyIndex);
     + item["Status"]["State"] = "Enabled";
```
And then 
L231:
```
      -  assemblyData["Status"]["State"] = value ? "Enabled" : "Absent";
     + if(!value) {
      +  assemblyData["Status"]["State"] = = "Absent";
     + }
```

Reply:
well,I can incorporate that logic in line 222. as we hit the line 222 only if some issue with fetching Present property.

Reply:
I'm okay if there is no default value and this logic is same as 1060.

Reply:
We tend not to use the ternary operator in bmcweb. Oh well

Reply:
I can see bmcweb has already used ternary operator https://github.com/openbmc/bmcweb/blob/master/redfish-core/include/utils/time_utils.hpp#L263 .. and do we have this anywhere documented?

Reply:
@baemyung yes, this logic covers the corner case which you mentioned. so I don't see any issues here.

Reply:
@PriyangaRamasamy 
Looking at the code again, "Enabed" is set if getProperty() fails.
```
if (ec)
        {
            assemblyData["Status"]["State"] = "Enabled";
...
            return;
        }
       assemblyData["Status"]["State"] = value ? "Enabled" : "Absent";
```

This implies that its default value is "Enabled".  
Then, I think the code is better to be done like as I suggested.

Set it as "Enabled" as default first.

```
     + if(!value) {
     +      assemblyData["Status"]["State"] = = "Absent";
     + }
```

Reply:
modified